c/tskit/tables.c

```diff
@@ -5498,6 +5498,92 @@ tsk_ibd_finder_init_samples(tsk_ibd_finder_t *self)
     return ret;
 }
 
+static int
+tsk_ibd_finder_index_samples(tsk_ibd_finder_t *self)
+{
+    int ret = 0;
+    size_t i;
+    tsk_id_t idx;
+
+    self->paired_nodes_index
+        = calloc(self->num_nodes, sizeof(*self->paired_nodes_index));
+
+    if (self->paired_nodes_index == NULL) {
+        ret = TSK_ERR_NO_MEMORY;
+        goto out;
+    }
+
+    for (i = 0; i < self->num_nodes; ++i) {
+        self->paired_nodes_index[i] = -1;
+    }
+
+    for (i = 0; i < 2 * self->num_pairs; ++i) {
+        self->paired_nodes_index[self->pairs[i]] = 1;
+    }
+
+    self->num_unique_nodes_in_pair = 0;
+    idx = 0;
+    for (i = 0; i < self->num_nodes; ++i) {
+        if (self->paired_nodes_index[i] == 1) {
+            self->num_unique_nodes_in_pair++;
+            self->paired_nodes_index[i] = idx++;
```

Comment:
We avoid using the return value of ``++`` - can you split this into two statements please?

Reply:
Just pushed a fix for this.  I can also squash this entire thing down to 1 commit?

Reply:
Squash away!